### ISSUE TITLE
Unable to connect to display :0 wsl2, ubuntu

### ISSUE THREAD

**Issue by pears0** [microsoft/wslg#902]
### Windows build number:

10.0.22621.0

### Your Distribution version:

20.04

### Your WSL versions:

WSL version: 1.0.0.0
Kernel version: 5.15.74.2
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.819

### Steps to reproduce:

firefox
Unable to init server: Could not connect: Connection refused
Error: cannot open display: :0

### WSL logs:

[stderr.log](https://github.com/microsoft/wslg/files/10082329/stderr.log)
[weston.log](https://github.com/microsoft/wslg/files/10082330/weston.log)
[pulseaudio.log](https://github.com/microsoft/wslg/files/10082331/pulseaudio.log)

### Expected behavior:

Firefox runs as a window inside of my windows 11 environment

### Actual behavior:

Failes to start

**Comment by pears0** [comment]
Additional useful information:

I've had to manually create the .X11-unix symlink, but the diurectory is empty


`╭─    ~ ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✔  2.5.8   at 08:47:39  ─╮
╰─ echo $DISPLAY                                                                                                                                                                         ─╯
:0
╭─    ~ ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✔  2.5.8   at 08:48:46  ─╮
╰─  ls -la /tmp/.X11-unix                                                                                                                                                                ─╯
total 4
drwxrwxrwt  2 root root   60 Nov 24 08:47 .
drwxrwxrwt 12 root root 4096 Nov 24 08:47 ..
lrwxrwxrwx  1 root root   19 Nov 24 08:47 .X11-unix -> /mnt/wslg/.X11-unix
╭─    ~ ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✔  2.5.8   at 08:48:59  ─╮
╰─ ls /tmp/.X11-unix                                                                                                                                                                     ─╯
╭─    ~ ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✔  2.5.8   at 08:49:10  ─╮
╰─ cat /mnt/wslg/versions.txt                                                                                                                                                            ─╯
WSLg ( x86_64 ): 1.0.47+Branch.main.Sha.77aa1e2f49e1253cde9ceb9124ca0873eb3eadf2
Mariner: VERSION="2.0.20221010"
DirectX-Headers:
mesa:
pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
FreeRDP: c574044a10003e50453acb4cf42801c5833fb572
weston: e7e91e17f370702f386c843cf9014cc4bcef4179`

**Comment by AlphaScorpii-dev** [comment]
Same issue on openSUSE-Tumbleweed. Oddly enough, I have a Debian and an Ubuntu installed as well, and /tmp is properly populated with the entries in /usr/lib/tmpfiles.d/x11.conf
The openSUSE instance was working fine just after installation, but updating the packages inside seems to have broken things => perhaps a problem with systemd-tmpfiles configuration ?

**Comment by crramirez** [comment]
I'll add this link here if someone finds this thread. https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/

It explains everything

**Comment by pears0** [comment]
> I'll add this link here if someone finds this thread. https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/
> 
> It explains everything

It explanes nothing about this issue to me or how to resolve it. It looks more like you're just plugging your blog.....

**Comment by crramirez** [comment]
You are right. I pasted in the wrong issue. And it is not my blog, it is the WSL official one. Thread others with respect if you want the same.

**Comment by pears0** [comment]
> 

Learn to take criticisem when you make a mistake and don't throw your toys out the pram when you are called out on it because you've wasted someones time with it.

**Comment by crramirez** [comment]
> > 
> 
> Learn to take criticisem when you make a mistake and don't throw your toys out the pram when you are called out on it because you've wasted someones time with it.

One thing is to take criticism for my mistake, which I did. And another very different is about: "It looks more like you're just plugging your blog" where:

1. You are accusing me of "plugging my blog" without facts.
2. You are lying telling me that this is my blog, again without facts.

And then you pretend to shut me up, insulting me by saying "throw your toys out the pram".

Who do you think you are accusing others without proof? Do you treat that way everyone that by mistake makes you waste time?

**Comment by pears0** [comment]
"Treat others with respect if you want the same"

You, when you earn it, you'll get it. But wasting time is a way to lose it, not earn it.

**Comment by Masamune3210** [comment]
Get over yourself, if you act with malice people are going to return malice as a result. They made a mistake, move on

**Comment by cybanical** [comment]
Hi, is there a solution to this? I followed "instructions" from MS site: https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps as well as installing the NVIDIA Cuda drivers,  and for example after installing GEDIT  I get error:

```
Unable to init server: Could not connect: Connection refused

(gedit:1211): Gtk-WARNING **: 00:40:58.222: cannot open display: :0
```

Also tried Nautilus with similar error
```
** (org.gnome.Nautilus:855): WARNING **: 00:40:26.980: Error on getting connection: Failed to load SPARQL backend: Error spawning command line ?dbus-launch --autolaunch=5530c00b1c29cde108c8b02463884c9a --binary-syntax --close-stderr?: Child process exited with code 1
Unable to init server: Could not connect: Connection refused

(org.gnome.Nautilus:855): Gtk-WARNING **: 00:40:26.986: cannot open display: :0
```

**Comment by kasvith** [comment]
Im also having same issue as @cybanical

**Comment by romanlex** [comment]
Hello, help for me
```
 sudo umount /tmp/.X11-unix
 sudo rm -R /tmp/.X11-unix
 ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix
 ```
 and now reboot wsl
 ```
 wsl --shutdown
 ```

**Comment by kasvith** [comment]
> Did you disable any option in .wslconfig or wsl.conf file?

Nope

**Comment by pears0** [comment]
> > Hello, help for me
> > ```
> >  sudo umount /tmp/.X11-unix
> >  sudo rm -R /tmp/.X11-unix
> >  ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and now reboot wsl
> > ```
> > wsl --shutdown
> > ```
> 
> My problem is wslg is not appearing in `/mnt/wslg`

This sound more like the WSGL instance hasn't started up for you, which is different to the issue that I was experiancing.

**Comment by romanlex** [comment]
> > > Hello, help for me
> > > ```
> > >  sudo umount /tmp/.X11-unix
> > >  sudo rm -R /tmp/.X11-unix
> > >  ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > and now reboot wsl
> > > ```
> > > wsl --shutdown
> > > ```
> > 
> > 
> > My problem is wslg is not appearing in `/mnt/wslg`
> 
> This sound more like the WSGL instance hasn't started up for you, which is different to the issue that I was experiancing.

I can run gnome-calculator but if I start chromium or another program I get `cannot open display: :0`
